tests/src/test/scala/org/apache/openwhisk/core/scheduler/queue/test/ContainerCounterTests.scala

```diff
@@ -247,11 +247,41 @@ class ContainerCounterTests
     NamespaceContainerCount.instances.clear()
   }
 
-  class MockEtcdClient(client: Client, isLeader: Boolean, leaseNotFound: Boolean = false, failedCount: Int = 1)
+  it should "update the number of containers correctly when multiple entries are inserted into etcd" in {
+    val mockEtcdClient = new MockEtcdClient(client, true, failedCount = 1)
+    val watcher = system.actorOf(WatcherService.props(mockEtcdClient))
+
+    val ns = NamespaceContainerCount(namespace, mockEtcdClient, watcher)
+    Thread.sleep(1000)
```

Comment:
I hope we avoid using this kind of waiting if possible.
Can we replace this with something like a reactive one?
For example, we might periodically check the expected status during the given time limit with some interval.

Reply:
replaced with retry, but there are many `Thread.sleep` in tests files, we may need an another pr to replace them

Reply:
Agree.
I just wanted not to add more `Thread.sleep`.